src/lib.rs

```diff
@@ -51,6 +51,31 @@ mod heapsize;
 
 // FIXME(conventions): implement indexing?
 
+/// The type returned by the `insert` function.
+#[derive(Clone)]
+pub struct Insert<K, V> {
+    pub old_value: Option<V>,
+    pub lru_removed: Option<(K, V)>,
+}
+
+impl<K: Eq + Hash, V: PartialEq> PartialEq for Insert<K, V> {
+    fn eq(&self, other: &Insert<K, V>) -> bool {
+        self.old_value == other.old_value
+        && self.lru_removed == self.lru_removed
```

Comment:
1. There is a typo in this line where both sides of the equality reference `self.  The latter should be `other`:
```
&& self.lru_removed == other.lru_removed
```
2. Could we derive this form of PartialEq? It is an element-by-element check for eq, which is the default for `#[derive(PartialEq)]`